### ISSUE TITLE
Kinematic bodies don't support total gravity

### ISSUE THREAD

**Issue by tofoz** [godot-jolt/godot-jolt#602]
when accessing a CharacterBody3D physics state to get its current gravity values it returns a vector of length zero. The problem does not exist for Godot's default physics.

Example in c#.
``` c#
        var state = PhysicsServer3D.BodyGetDirectState(body.GetRid());
        GravityDirection = state.TotalGravity.Normalized();
        GravityScale = state.TotalGravity.Length();
```

**Comment by mihe** [comment]
I don't believe you're supposed to be able to use `PhysicsDirectBodyState3D` with `CharacterBody3D`, hence why you're forced to go through the physics server to actually get it in the first place.

As noted in [its documentation](https://docs.godotengine.org/en/stable/classes/class_physicsdirectbodystate3d.html):

> This object is passed via the direct state callback of [RigidBody3D](https://docs.godotengine.org/en/stable/classes/class_rigidbody3d.html#class-rigidbody3d), and is intended for changing the direct state of that body.

I can however see how being able to get the total/accumulated gravity and total/accumulated damping can be useful when walking through areas with a `CharacterBody3D`, but I would need to confirm that I'm not just replicating some unintended (and potentially buggy) behavior from Godot Physics.

Maybe @rburing would have some idea?

**Comment by mihe** [comment]
Hm, alright, this seems to be supported by Bullet in Godot 3 as well, and seems to have made its way into the [Godot XR template](https://github.com/GodotVR/godot-xr-template/blob/43266aeaf82be5755a0053ef53eec1a6021b7112/addons/godot-xr-tools/player/player_body.gd#L251-L252) as well. I guess just adding support for it is the better option then.

**Comment by tofoz** [comment]
@mihe Both I have a character controller that can have its up orientation override and was using an Area3d with gravity_space_override to do so. I just tested and could just use a rigid body as I am only using MoveAndCollide for the controller.

**Comment by mihe** [comment]
> I just tested and could just use a rigid body as I am only using MoveAndCollide for the controller.

I would expect that to behave quite oddly with a normal `RigidBody3D`. For it to work properly with `MoveAndCollide` you would likely need to freeze the body with the "Kinematic" freeze mode, and in that case you would run into the same problem as with `CharacterBody3D`.

I'll see about adding support for it. It should be a quick fix I think.

**Comment by mihe** [comment]
The fix for this is included in the newly released [0.8.0-rc1](https://github.com/godot-jolt/godot-jolt/releases/tag/v0.8.0-rc1). If you're able to, I'd appreciate it if you could give that release a try and report back whether it solves the issue for you.

**Comment by mihe** [comment]
The fix for this is now also included in the newly released [0.8.0-stable](https://github.com/godot-jolt/godot-jolt/releases/tag/v0.8.0-stable).
